packages/vkui/src/components/AppRoot/AppRootPortal.tsx

```diff
@@ -2,62 +2,92 @@
+import { AppRootStyleContainer } from './AppRootStyleContainer';
 
 export interface AppRootPortalProps extends HasChildren {
   /**
    * - При передаче `true` будет использовать `portalRoot` из контекста `AppRoot`.
    * - При передаче элемента будут игнорироваться `portalRoot` и `disablePortal` из контекста `AppRoot`.
+   * - При передаче `in-app-after-content` будет использоваться контейнер внутри `SplitLayout`, сразу после контента приложения.
    */
-  usePortal?: boolean | HTMLElement | React.RefObject<HTMLElement> | null;
+  usePortal?: boolean | HTMLElement | React.RefObject<HTMLElement> | null | 'in-app-after-content';
+  className?: string;
 }
 
-export const AppRootPortal = ({ children, usePortal }: AppRootPortalProps): React.ReactNode => {
-  const { portalRoot, mode, disablePortal } = React.useContext(AppRootContext);
+export const AppRootPortal = ({
+  children,
+  usePortal,
+  className,
+}: AppRootPortalProps): React.ReactNode => {
+  const { setPortalRoot, appRoot, mode, disablePortal } = React.useContext(AppRootContext);
   const appearance = useAppearance();
 
-  const isClient = useIsClient();
-  if (!isClient) {
-    return null;
-  }
+  const canUsePortal = shouldUsePortal(usePortal, mode, Boolean(disablePortal));
+  const portalContainer = usePortalContainer(usePortal);
 
-  const portalContainer = resolvePortalContainer(usePortal, portalRoot.current);
-  if (!portalContainer || shouldDisablePortal(usePortal, mode, Boolean(disablePortal))) {
-    return <React.Fragment>{children}</React.Fragment>;
-  }
+  useIsomorphicLayoutEffect(
+    // Создаём контейнер для портала по запросу один раз
+    // и пока приложение не размонтируется.
+    // Удаление созданной ноды происходит в AppRoot
+    function initializePortalRootIfNeeded() {
+      const shouldCreatePortalRoot = canUsePortal && portalContainer === null;
+      if (shouldCreatePortalRoot) {
+        const documentBody = getDocumentBody(appRoot.current);
+        const portal = documentBody.ownerDocument.createElement('div');
+        documentBody.appendChild(portal);
 
-  return createPortal(
-    <AppearanceProvider value={appearance}>{children}</AppearanceProvider>,
-    portalContainer,
+        setPortalRoot(portal);
+      }
+
+      // делать очистку и удалять portalRoot не нужно,
+      // так как это произойдёт при размонтировании AppRoot
+    },
+    [canUsePortal, appRoot, portalContainer, setPortalRoot],
   );
+
+  if (canUsePortal && portalContainer) {
+    return createPortal(
+      <AppearanceProvider value={appearance}>
+        <AppRootStyleContainer className={className}>{children}</AppRootStyleContainer>
+      </AppearanceProvider>,
+      portalContainer,
+    );
+  }
+
+  return children;
 };
 
-function shouldDisablePortal(
+function shouldUsePortal(
   usePortal: AppRootPortalProps['usePortal'],
   mode: AppRootContextInterface['mode'],
   disablePortal: boolean,
```

Comment:
```suggestion
  globalDisablePortalProp: boolean,
```
давай назовём как-то более говоряще, а то сейчас выглядит как параметр противоположный `useProtal`

Reply:
> давай назовём как-то более говоряще, а то сейчас выглядит как параметр противоположный useProtal

Понимаю, подумаю над названием 🤔